src/common/common.go

```diff
@@ -132,3 +132,36 @@ func IsK8sNodeIsReady(node v1.Node) bool {
 	}
 	return false
 }
+
+// BuildHostsMapIPAddressBased builds a map containing all the IP addresses of the nodes in the
+// inventory so that later we can match reporting nodes based on the IP and not only on the name.
+func BuildHostsMapIPAddressBased(inventoryHostsMap map[string]inventory_client.HostData) map[string]inventory_client.HostData {
+	knownIpAddresses := map[string]inventory_client.HostData{}
+	for _, v := range inventoryHostsMap {
+		for _, ip := range v.IPs {
+			knownIpAddresses[ip] = v
```

Comment:
Is `v` a reference?
I would be a bit wary of using the loop variable here.

Maybe add some tests for this to make sure it's working the way you want.

Reply:
I think this is not an issue because you're not taking the address of the loop variable, but would still be good to have tests.

Reply:
Tests created